### PR TITLE
Add dockerfile for flower client image

### PR DESCRIPTION
**Pull request by Robert-Steiner**
## Issue

We currently offer docker images for our server but not for our client. In order to make developing with Flower as easy as possible, we want to close this gap. 

### Description

- adds dockerfile for flower client image

### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.dev/docs/writing-documentation.html)
- [ ] Update [changelog](https://github.com/adap/flower/blob/main/doc/source/changelog.rst)
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.dev/join-slack/) (channel `#contributions`)

### CODE REVIEW

File: src/docker/client/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Copyright 2023 Flower Labs GmbH. All Rights Reserved.
+
+ARG BASE_REPOSITORY=flwr/base
+ARG BASE_IMAGE_TAG
+FROM $BASE_REPOSITORY:$BASE_IMAGE_TAG
+
+ARG FLWR_VERSION
+RUN python -m pip install -U --no-cache-dir flwr"[rest, simulation]"==${FLWR_VERSION}
```
Comment (Robert-Steiner) [inline src/docker/client/Dockerfile:8]: @tanertopal I'm unsure, if we need the `simulation` feature

Comment (tanertopal) [reply src/docker/client/Dockerfile:8]: @danieljanes @jafermarq Not sure either. Do we need "simulation"
 in the client image? I would assume no but please correct me if I am wrong.

Comment (jafermarq) [reply src/docker/client/Dockerfile:8]: we don't

Comment (Robert-Steiner) [reply src/docker/client/Dockerfile]: I removed it

**Review by tanertopal — APPROVED** [review]

**Pull request merged**